### PR TITLE
Set up basic scaffold for database and model.

### LINKED ISSUE #3: TCR: add db schema
hints: 

- work with Julian
- which fields we need per requirements (e.g. audit fixity checking and archiving)

TCR will contain:
- checksums for manifestInventory.xml for EACH VERSION of each online Moab
- checksums for EACH VERSION of archived object (e.g. druid-version.zip)

6 tables:

ManifestInventoryMd5
ManifestInventorySha1
ManifestInventorySha256
ArchivalMd5
ArchivalSha1
ArchivalSha256

each table has columns 'druid', 'version', 'checksum'

fullsizeoutput_8a

### LINKED ISSUE #5: Create Rake task for Travis database setup
Once we have set up a database schema, we should add any code required to initialize the database both for Travis and for new developers. See e.g.

https://github.com/sul-dlss/preservation_core_catalog/blob/master/Rakefile#L13-L15

### PR DESCRIPTION
**Pull request by tingulfsen**
Fixes #3 

Fixes #5

This PR sets up a basic database table that holds the checksums as separate columns. There is one column per checksum, per object type:

- moab_md5
- moab_sha1
- moab_sha256
- archive_md5
- archive_sha1
- archive_sha256

The combination of a druid and version is always unique and there is an index on these columns.

Note that final database permissions and actual code is not included in this PR - that is for later tickets.

### CODE REVIEW

File: app/controllers/checksum_records_controller.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    else
+      render json: @checksum_record.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /checksum_records/1
+  # DELETE /checksum_records/1.json
+  def destroy
+    @checksum_record.destroy
+  end
```
Comment (ndushay) [inline app/controllers/checksum_records_controller.rb:41]: I think we don't want destroy per "specs"

File: app/controllers/checksum_records_controller.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    if @checksum_record.save
+      render :show, status: :created, location: @checksum_record
+    else
+      render json: @checksum_record.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /checksum_records/1
+  # PATCH/PUT /checksum_records/1.json
+  def update
```
Comment (ndushay) [inline app/controllers/checksum_records_controller.rb:29]: I think we don't want "update" per "specs"

File: config/routes.rb [hunk 3]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
+  resources :checksum_records
```
Comment (ndushay) [inline config/routes.rb:2]: we don't want all actions, only create and read, yes?  (maybe index?)

File: spec/routing/checksum_records_routing_spec.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    it "routes to #update via PUT" do
+      expect(:put => "/checksum_records/1").to route_to("checksum_records#update", :id => "1")
+    end
+
+    it "routes to #update via PATCH" do
+      expect(:patch => "/checksum_records/1").to route_to("checksum_records#update", :id => "1")
+    end
+
+    it "routes to #destroy" do
```
Comment (ndushay) [inline spec/routing/checksum_records_routing_spec.rb:28]: nope!

File: spec/routing/checksum_records_routing_spec.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+    it "routes to #create" do
+      expect(:post => "/checksum_records").to route_to("checksum_records#create")
+    end
+
+    it "routes to #update via PUT" do
+      expect(:put => "/checksum_records/1").to route_to("checksum_records#update", :id => "1")
+    end
+
+    it "routes to #update via PATCH" do
```
Comment (ndushay) [inline spec/routing/checksum_records_routing_spec.rb:24]: nope! (?)

File: spec/routing/checksum_records_routing_spec.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    it "routes to #show" do
+      expect(:get => "/checksum_records/1").to route_to("checksum_records#show", :id => "1")
+    end
+
+
+    it "routes to #create" do
+      expect(:post => "/checksum_records").to route_to("checksum_records#create")
+    end
+
+    it "routes to #update via PUT" do
```
Comment (ndushay) [inline spec/routing/checksum_records_routing_spec.rb:20]: nope! (?).  only create, show and possibly index?

**Review by ndushay — CHANGES_REQUESTED** [review]
I don't think we want all the actions enabled in this PR, do we?

Also wondering about a model with appropriate validations ...

File: .rubocop_todo.yml [hunk 7]
```diff
@@ -20,14 +20,40 @@ Layout/CommentIndentation:
   Exclude:
     - 'config/environments/production.rb'
```
Comment (ndushay) [inline .rubocop_todo.yml:22]: I think you'll find, for rubocop, that there are some reasonable patterns in other rails apps of files to exclude from rubocop (e.g. the autogenerated ones, etc.)

**Review by ndushay — CHANGES_REQUESTED** [review]

File: app/controllers/checksum_records_controller.rb (hunk 2 above)
Comment (SaravShah) [reply app/controllers/checksum_records_controller.rb:29]: https://docs.google.com/document/d/18MPHfUtEwNrCp2idmSt-TvDnxF3P0Kt3WWWnN5pBZ7M/edit#heading=h.yzqsqntox27w

File: .rubocop_todo.yml (hunk 7 above)
Comment (ndushay) [reply .rubocop_todo.yml:22]: I just created issue #14 to address rubocop settings.

File: app/controllers/checksum_records_controller.rb (hunk 2 above)
Comment (ndushay) [reply app/controllers/checksum_records_controller.rb:29]: Tommy created issue #11 to deal with this.

File: app/controllers/checksum_records_controller.rb (hunk 1 above)
Comment (ndushay) [reply app/controllers/checksum_records_controller.rb:41]: Tommy created issue #11 to deal with this.

File: config/routes.rb (hunk 3 above)
Comment (ndushay) [reply config/routes.rb:2]: Tommy created issue #11 to deal with this.

File: spec/routing/checksum_records_routing_spec.rb (hunk 6 above)
Comment (ndushay) [reply spec/routing/checksum_records_routing_spec.rb:20]: Tommy created issue #11 to deal with this.

File: spec/routing/checksum_records_routing_spec.rb (hunk 5 above)
Comment (ndushay) [reply spec/routing/checksum_records_routing_spec.rb:24]: Tommy created issue #11 to deal with this.

File: spec/routing/checksum_records_routing_spec.rb (hunk 4 above)
Comment (ndushay) [reply spec/routing/checksum_records_routing_spec.rb:28]: Tommy created issue #11 to deal with this.

**Review by ndushay — APPROVED** [review]

File: db/scripts/db_setup.sql [hunk 8]
```diff
@@ -0,0 +1,11 @@
+CREATE USER tcr;
+ALTER USER tcr CREATEDB;
+CREATE DATABASE tcr;
+ALTER DATABASE tcr OWNER TO tcr;
+GRANT ALL PRIVILEGES ON DATABASE tcr TO tcr;
```
Comment (jmartin-sul) [inline db/scripts/db_setup.sql:5]: was going to say that we probably don't want to grant all privs, but looks like that comment's already captured in #10

File: app/controllers/checksum_records_controller.rb [hunk 9]
```diff
@@ -0,0 +1,53 @@
+class ChecksumRecordsController < ApplicationController
+  before_action :set_checksum_record, only: [:show, :update, :destroy]
+
+  # GET /checksum_records
+  # GET /checksum_records.json
+  def index
+    @checksum_records = ChecksumRecord.all
+  end
```
Comment (jmartin-sul) [inline app/controllers/checksum_records_controller.rb:8]: we might also want to disallow listing all records, since there might be lots.  filed #12.

**Review by jmartin-sul — APPROVED** [review]
filed a couple issues, and approving/merging, in the spirit of getting the ball rolling and cleaning up rough edges as we continue to implement.

**Pull request merged**